### PR TITLE
Replace solve with solveset in core

### PR DESCRIPTION
**Pull request by aktech**
TODO:
- [x] `expr.py`
- [x] `test_function.py`
- [x] `test_wester.py`

@hargup  @flacjacket 

Partial fix for #8711

### CODE REVIEW

File: sympy/core/tests/test_wester.py [hunk 1]
```diff
@@ -924,56 +929,67 @@ def test_M12():
...
 
 def test_M22():
-    assert solve(2*sqrt(x) + 3*x**R(1, 4) - 2) == [R(1, 16)]
+    assert solveset(2*sqrt(x) + 3*x**R(1, 4) - 2) == FiniteSet(R(1, 16))
 
 
 def test_M23():
     x = symbols('x', complex=True)
-
+    # TODO: Replace solve with solveset, as of now test fails for solveset
```
Comment (hargup) [inline sympy/core/tests/test_wester.py]: You should open an issue for this.

File: sympy/core/tests/test_wester.py [hunk 2]
```diff
@@ -1163,55 +1199,55 @@ def test_N8():
...
 @XFAIL
 def test_N12():
     x = Symbol('x', real=True)
-    assert solve(sqrt(x) < 2) == And(Le(0, x), Lt(x, 4))
+    assert solveset(sqrt(x) < 2) == Interval(0, 4, False, True)
 
 
 @XFAIL
 def test_N13():
     # raises NotImplementedError: can't reduce [sin(x) < 2]
```
Comment (hargup) [inline sympy/core/tests/test_wester.py]: You should also remove this comment.

File: sympy/core/tests/test_wester.py [hunk 3]
```diff
@@ -924,56 +929,67 @@ def test_M12():
...
 def test_M16():
-    assert solve(sin(x) - tan(x), x) == [0, -pi, pi, 2*pi]
+    n = Dummy('n')
+    assert solve(sin(x) - tan(x), x) == ImageSet(Lambda(n, n*pi), Integers())
 
 
 @XFAIL
 def test_M17():
-    assert solve(asin(x) - atan(x),x) == [0]
+    assert solveset_real(asin(x) - atan(x), x) == FiniteSet(0)
```
Comment (hargup) [inline sympy/core/tests/test_wester.py]: extra whitespace.

Comment (aktech) [inline sympy/core/tests/test_wester.py]: where?

Comment (hargup) [inline sympy/core/tests/test_wester.py]: After the comma.

Comment (aktech) [inline sympy/core/tests/test_wester.py]: that's only 1 white-space, isn't it?

Comment (hargup) [inline sympy/core/tests/test_wester.py]: diff

Comment (aktech) [inline sympy/core/tests/test_wester.py]: There should be a white space after a comma, earlier it wasn't so I had put that.

If I would remove that, the PEP8 error would be:
`missing whitespace after ','`

Comment (hargup) [inline sympy/core/tests/test_wester.py]: okay.

**Comment by hargup** [comment]
+1

**Comment by moorepants** [comment]
I don't think the solve tests should have been removed here.

**Comment by hargup** [comment]
@aktech you should revert the last commit and restore `solve` tests.

**Pull request merged**